pkg/server/endpoints/endpoints.go

```diff
@@ -289,65 +288,35 @@ func (e *Endpoints) runLocalAccess(ctx context.Context, server *grpc.Server) err
 // getTLSConfig returns a TLS Config hook for the gRPC server
 func (e *Endpoints) getTLSConfig(ctx context.Context) func(*tls.ClientHelloInfo) (*tls.Config, error) {
 	return func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
-		certs, roots, err := e.getCerts(ctx)
-		if err != nil {
-			e.Log.WithError(err).WithField(telemetry.Address, hello.Conn.RemoteAddr().String()).Error("Could not generate TLS config for gRPC client")
-			return nil, err
-		}
-
-		return &tls.Config{
-			// Not all server APIs required a client certificate. Though if one
-			// is presented, verify it.
-			ClientAuth: tls.VerifyClientCertIfGiven,
-
-			Certificates: certs,
-			ClientCAs:    roots,
-
-			MinVersion: tls.VersionTLS12,
-
-			NextProtos: []string{http2.NextProtoTLS},
-		}, nil
+		serverSvid := newX509SVIDSource(e.getServerCertificate)
+		bundle := newBundleSource(func(td *spiffeid.TrustDomain) ([]*x509.Certificate, error) {
+			return e.bundleGetter(ctx, td)
+		})
+
+		spiffeTLSConfig := tlsconfig.MTLSServerConfig(serverSvid, bundle, nil)
+		spiffeTLSConfig.ClientAuth = tls.RequestClientCert
```

Comment:
Can we leave a note here just emphasizing that client certificates will still be verified using the custom VerifyPeerCertificates callback?

Reply:
sure!